q2_types/per_sample_sequences/tests/test_format.py

```diff
@@ -26,6 +28,97 @@
 from qiime2.plugin import ValidationError
 
 
+class TestAbsoluteFastqManifestV2Formats(TestPluginBase):
+    package = 'q2_types.per_sample_sequences.tests'
+
+    def setUp(self):
+        super().setUp()
+        self.se_formats = [SingleEndFastqManifestPhred33V2,
+                           SingleEndFastqManifestPhred64V2]
+        self.pe_formats = [PairedEndFastqManifestPhred33V2,
+                           PairedEndFastqManifestPhred64V2]
+
+    def template_manifest(self, filepath, ctx):
+        with open(filepath) as fh:
+            tmpl = string.Template(fh.read())
+        basename = os.path.basename(filepath)
+        file_ = os.path.join(self.temp_dir.name, basename)
+        with open(file_, 'w') as fh:
+            fh.write(tmpl.substitute(**ctx))
+        return file_
+
+    def test_validate_se_positive(self):
+        s1 = self.get_data_path('Human-Kneecap_S1_L001_R1_001.fastq.gz')
+        fp = self.get_data_path('absolute_manifests_v2/single-MANIFEST')
+        manifest = self.template_manifest(fp, {'path': s1})
+
+        for fmt in self.se_formats:
+            fmt(manifest, mode='r').validate()
+
+    def test_validate_pe_positive(self):
+        s1 = self.get_data_path('Human-Kneecap_S1_L001_R1_001.fastq.gz')
+        fp = self.get_data_path('absolute_manifests_v2/paired-MANIFEST')
+        manifest = self.template_manifest(fp, {'path': s1})
+
+        for fmt in self.pe_formats:
+            fmt(manifest, mode='r').validate()
+
+    def test_extra_columns(self):
+        s1 = self.get_data_path('Human-Kneecap_S1_L001_R1_001.fastq.gz')
+        fp = self.get_data_path('absolute_manifests_v2/multicol-MANIFEST')
+        manifest = self.template_manifest(fp, {'path': s1})
+
+        for fmt in self.se_formats:
+            fmt(manifest, mode='r').validate()
+
+    def test_invalid_metadata(self):
+        manifest = self.get_data_path('absolute_manifests/single-MANIFEST')
+
+        for fmt in self.se_formats:
+            with self.assertRaisesRegex(ValidationError, 'unrecognized ID'):
+                fmt(manifest, mode='r').validate()
+
+    def test_missing_column_se(self):
+        manifest = self.get_data_path('absolute_manifests_v2/paired-MANIFEST')
+
+        for fmt in self.se_formats:
+            with self.assertRaisesRegex(ValidationError, 'is not a column'):
```

Comment:
It seems to me like the error message should be that this column is missing from the metadata, but I'm not seeing where this error gets raised so I'm not sure what the full error message says.

Reply:
https://github.com/qiime2/qiime2/blob/af5bdca1326ede6d6f071573a3c9f18687c5ac58/qiime2/metadata/metadata.py#L537-L539

Reply:
perfect, thanks!